### ISSUE TITLE
Objects in page.txt

### ISSUE THREAD

**Issue by gsolina** [datenstrom/yellow#134]
Dear Sirs
Sorry I am a beginner.
I put the following in one of the page.txt file

<?php echo $yellow->config->getHtml("ServerName") ?>

but I do not get anything printed.

ServerName is set in config.ini file as www.servername.hello

Can you please help me? What's wrong with my php and page.txt ?

Thank you for your cooperation

Best regards, Giuseppe

**Comment by wunderfeyd** [comment]
The content files parsed by a parser (Markdown as default) and your php code isn't interpreted. To add php code you have to modify [snippets or templates](https://github.com/datenstrom/yellow/wiki/Yellow-customisation).

`ServerName` is needed if you want to build static sites. Do you want to do this?

What do you try to achieve?

**Comment by gsolina** [comment]
Hello.
I do not want to build a static site.
I would like to read some settings from the config.ini file (for example: ContactEmail, ContactTelephone etc) and those values in the middle of a page, between the html code which I usually write in the page.txt file.
Like this...

---
Title: A longer title for the page
TitleNavigation: A short title for the navigation
description: Some words about me
keywords: me, myself, gsolina
author: gsolina
language: en
---
<h1>This is the tile of my page</h1>
<p>This is a paragraph of my page</p>
<a href="http://www.somedomain.com" target="_blank">This is a link of my page</a>
<p>I would like to use php code here...</p>
<?php echo $yellow->config->getHtml("ContactEmail") ?>
<?php echo $yellow->config->getHtml("ContactTelephone") ?>

Thank you very much for your cooperation 

Giuseppe

**Comment by wunderfeyd** [comment]
The normal way is to use the [templates and snippets](
https://github.com/datenstrom/yellow/wiki/Yellow-customisation#custom-template). But you can go the other way by creating a plugin which supports you to embed php code.

```php
<?php
// Copyright (c) 2013-2015 Datenstrom, http://datenstrom.se
// This file may be used and distributed under the terms of the public license.

// Translation plugin
class YellowTranslation
{
	const Version = "0.6.1";
	var $yellow;			//access to API

	// Handle initialisation
	function onLoad($yellow)
	{
 		$this->yellow = $yellow;
	}

	// Handle page content parsing of custom block
	function onParseContentBlock($page, $name, $text, $shortcut)
	{
		$output = NULL;
		if($name=="translate" && $shortcut)
		{
			list($key) = $this->yellow->toolbox->getTextArgs($text);
			$output = $this->yellow->text->getHtml(lcfirst($key));			
		 } else if($name=="tag" && $shortcut) {
			list($key) = $this->yellow->toolbox->getTextArgs($text);
			$output = $page->getHtml(lcfirst($key));		
		} else if($name=="setting" && $shortcut) {
			list($key) = $this->yellow->toolbox->getTextArgs($text);
			$output = $this->yellow->config->getHtml(lcfirst($key));
		}
		return $output;
	}
}

$yellow->plugins->register("translation", "YellowTranslation", YellowTranslation::Version);
?>
```

Store this code in the `system/plugins` folder and name it for example `translate.php`. If done you can access tags, settings and language texts.

Example:
```
---
Title: Test
---
[setting Sitename]
[setting ServerName]
[translate BlogMore]
[tag Title]
``` 

Note this is just an example code (it's not tested well and there's no guaranty it's safe and working as expected). Feel free to modify and add your features and/or php code.

**Comment by wunderfeyd** [comment]
Did this help? Have you found another solution?

**Comment by gsolina** [comment]
Hello everybody
Sorry for the late replay. I was not at office.
I will check late by today and revert to you asap.
Thank you very much for your cooperation 
Best regards
Giuseppe

**Comment by gsolina** [comment]
Hello wunderfeyd
Thank you very much for your cooperation.
I named your code as translate.php and stored in the system/plugins folder.
Then I stored a page.txt file like this...
```
---
Title: My Page
---
Your website works!

You can now [edit this page] or use your text editor.

<a href="[setting ServerName]">This is a link to the ServerName</a>

[setting ServerName]

[tag Title]
```
Unfortunately the link `<a href>` does not work.
So I can access tags, settings and language texts but I do not know how I can use them inside html code.
Thank you for your cooperation

Best regards, Giuseppe

PS: I apologize for the late replay. I was not at office in the past days.

**Comment by wunderfeyd** [comment]
When referencing the server itself, you should use the `/` (slash). Since it does the same as the reference to the same domain name. Use relative paths as long as possible to avoid other kind of strange problems.

Yes it's right, the plugin might not work inside other Markdown elements and embedded html elements. This is because the method used is a callback from the Markdown parser. 

You can put the whole html you would like to create into the `translate.php` and use your own tags or you could create a plugin which replaces you definied tags before/after parsing of Markdown. You can for example adapt https://github.com/varakh/yellowcms-extensions-replacetext/blob/master/replacetextfa.php with your own extentions.

If it's just the server name (as stated above) go the simple path. Please have a look into the [templates and snippets](
https://github.com/datenstrom/yellow/wiki/Yellow-customisation#custom-template) documentation, I'm very sure it would fulfill your needs already, if you understand what's happening behind the scene.

These are just some solutions if I understood your problem/ticket correctly.

**Comment by markseu** [comment]
Just ask if there are more questions. :smile:

**Comment by gsolina** [comment]
Dear
Thank you very much for your effort.
Unfortunately this subject is getting too complicated for me. I am not a professional web developer.
If I could find one, I would be very happy to pay is bill.
Unfortunately I met a (so called) web developer 6 moths ago and what I got is a web site nice-looking (maybe) but not working.
The web site is based on Yellow but has got mistakes. For example after 6 months the contact form is not sending email.
I tried to get into the subject by myself but I do not know enough about web developing. It is not my job.
I would just need a serious web developer.
Sorry I am frustrated.
Thanks for your time
Best regards
Giuseppe
